services/orchest-webserver/app/app/core/filemanager.py

```diff
@@ -30,30 +30,28 @@ def _construct_root_dir_path(
     project_uuid: Optional[str],
     pipeline_uuid: Optional[str] = None,
     job_uuid: Optional[str] = None,
-    run_uuid: Optional[str] = None,
+    run_uuid_or_snapshot: Optional[str] = None,
 ) -> str:
     """
     If root is not provided, default to PROJECT_DIR_PATH;
     If root equals to PROJECT_DIR_PATH, project_uuid is required;
     """
 
     root = PROJECT_DIR_PATH if root is None else root
-
     if root not in ALLOWED_ROOTS:
         raise ValueError(f"Received illegal root path: {root}.")
 
     if root == DATA_DIR_PATH:
         return _config.USERDIR_DATA
-
-    if root == PROJECT_DIR_PATH and project_uuid is None:
+    elif project_uuid is None:
         raise ValueError("project_uuid is required.")
-
-    return get_project_directory(
-        project_uuid=project_uuid,
-        pipeline_uuid=pipeline_uuid,
-        job_uuid=job_uuid,
-        run_uuid=run_uuid,
-    )
+    else:
+        return get_project_directory(
+            project_uuid=project_uuid,
+            pipeline_uuid=pipeline_uuid,
+            job_uuid=job_uuid,
+            run_uuid_or_snapshot=run_uuid_or_snapshot,
```

Comment:
Would it be possible to pass these individually instead? :thinking: 

```suggestion
            run_uuid=run_uuid,
            snapshot_uuid=snapshot_uuid,
```

Reply:
`run_uuid_or_snapshot` doesn't contain anything about `snapshot_uuid`. This variable is either a run UUID or a string literal "snapshot".  🙂

Reply:
Oh! Haha. :facepalm:

It can be left as-is then. :+1: